### PR TITLE
Unify placeholders

### PR DESCRIPTION
**Pull request by jasmussen**
## What?

Update and unify placeholder styles. This makes the Columns block have visually the same setup as Group does. Also updates Image, Video, Audio and other placeholders, emphasizing compactness so that they scale better to narrow contexts.

placeholders latest appearance

<details><summary>Previous iteration of this description</summary>

WIP exploration to explore unifying the Placeholder component. 

Before:

placeholders-before

After:

placeholders-after

</details>

### Testing Instructions for Keyboard

Test blocks that have setup states; group, columns, query loop, image, etc.

### CODE REVIEW

**Comment by fabiankaegy** [comment]
Whilst I appreciate the minimalistic visual of this I am converned about the usibility of this. Loosing the clear heading / label for which block the placeholder state is for is a big regression for me as you get lost much more easily. 

I'm in favor of retaining the title in the placeholder.

I also just want to call out that changing the styling of the variation icons will cause any 3rd party variations to look out of place till they are updated. That is not a big deal but may cause an odd interim step for many.

**Comment by jasmussen** [comment]
Yep, definitely agree. This PR is a bit in provocation territory, exploring the "what if". I think there are actually pieces to this that are worth bringing forward, mainly that the code is such a tangled and fragmented mess at the moment. But also that the provocation here can hopefully serve as a catalyst to find the right configuration. E.g. integrating the title in a smaller footprint, and darkening the variation icon color, would be a trivial change, and could potentially still accomplish the same.

**Comment by jasmussen** [comment]
Quick sketch, which could use better verbiage but still feels more scalable:

Frame 3

**Comment by jasmussen** [comment]
I've updated this one to clean things up, include the label, and make it ready for review. I think this is a good step forward, let me know!

placeholders latest appearance

**Comment by jameskoster** [comment]
Seems like a step in the right direction.

Do you think we need the 'Skip' link for columns?

There's a quirk in the hover interaction – you can cursor over the button, and the background changes. But the icon doesn't change until you cursor further inside the button:


https://github.com/WordPress/gutenberg/assets/846565/a58fbdc9-bc68-4b44-9461-fc17e9e70d7c

**Comment by jasmussen** [comment]
> Do you think we need the 'Skip' link for columns?

I don't, but I kept it for now to keep the PR smallish, there's a good amount of red in the diff already.

Good catch on the hover interaction! My instinct is to make the icon _itself_ change color on hover, I'll do that unless anyone objects.

**Comment by jameskoster** [comment]
Not for this PR, and I know we discussed it before, but I wonder if the thumbnail 'icons' could be inspiration for the de-selected state for empty groups and columns, so that the UX matches images. For example an empty group could be a box, empty columns could be 3 adjacent boxes. Might work better in the site view?
| Updated placeholders | Current placeholders |
| --- | --- |
| Screenshot 2024-03-25 at 09 28 33 | Screenshot 2024-03-25 at 09 30 16 |

File: packages/block-editor/src/components/block-variation-picker/content.scss [hunk 1]
```diff
@@ -1,89 +1,32 @@
...
-		// Allow options to occupy a greater amount of the available space if
-		// many options exist.
-		max-width: 90%;
-	}
-}
-
-.block-editor-block-variation-picker__variations.block-editor-block-variation-picker__variations {
+.block-editor-block-variation-picker__variations,
+.block-editor-block-variation-picker__skip,
+.wp-block-group-placeholder__variations {
```
Comment (t-hamano) [inline packages/block-editor/src/components/block-variation-picker/content.scss:3]: I'm wondering if I could remove the `.wp-block-group-placeholder__variations` selector. This means using a true variation picker in [this part](https://github.com/WordPress/gutenberg/blob/ffc02bf54720ec95f3380d422e1cb44458e5a621/packages/block-library/src/group/placeholder.js#L148-L154) of the Group block.

File: packages/block-editor/src/components/block-variation-picker/content.scss [hunk 2]
```diff
@@ -1,89 +1,32 @@
...
-			display: inline-flex;
-			margin-right: 0;
-		}
+	svg {
+		fill: $gray-400 !important;
 	}
 
-	.block-editor-block-variation-picker__variation {
-		padding: $grid-unit-10;
+	:hover {
```
Comment (t-hamano) [inline packages/block-editor/src/components/block-variation-picker/content.scss]: What element is this selector targeting?

File: packages/block-editor/src/components/block-variation-picker/index.js [hunk 3]
```diff
@@ -53,9 +53,6 @@ function BlockVariationPicker( {
 							className="block-editor-block-variation-picker__variation"
 							label={ variation.description || variation.title }
 						/>
-						<span className="block-editor-block-variation-picker__variation-label">
-							{ variation.title }
-						</span>
```
Comment (t-hamano) [inline packages/block-editor/src/components/block-variation-picker/index.js]: I think this is a breaking change in the `BlockVariationPicker` component. The `title` prop defined in the block variation will no longer be referenced.

**Review by t-hamano — COMMENTED** [review]
If this PR ships, it will also resolve some of the issues mentioned in #41961.

In order to move this PR forward, I think it is necessary to investigate from two perspectives.

### Placeholder as a component

This PR brings major changes to the layout of the component. Below is a screenshot from Storybook.

Before:

before-storybook

After:

after-storybook

In my experience, this component is relatively popular among developers. I think we need to get feedback from @WordPress/gutenberg-components on how much of an impact this change will have.

### Placeholder as the initial state of the block

Regarding variation labels, I would recommend displaying them. There should be enough space to display the text.

For example, with the Column block, We won't be able to see the visual proportions of each variation.

column-placeholder

Adding visible text to variations is also mentioned in #60217.

**Comment by jasmussen** [comment]
Thank you for the review, I will try and address your points when I have a moment. Also feel free to push directly if you have chnages in the mean time. 🙏

File: packages/block-editor/src/components/block-variation-picker/content.scss (hunk 2 above)
Comment (richtabor) [reply packages/block-editor/src/components/block-variation-picker/content.scss:18]: It was targeting this: 

CleanShot 2024-04-15 at 17 24 23

Comment (richtabor) [reply packages/block-editor/src/components/block-variation-picker/content.scss]: I added .components-button.

File: packages/block-editor/src/components/block-variation-picker/index.js (hunk 3 above)
Comment (richtabor) [reply packages/block-editor/src/components/block-variation-picker/index.js]: It'll be referenced if there is no description. As an alternative, would adding a prop to not have title be fine for each variation? They're not always helpful, like in these.

Comment (richtabor) [reply packages/block-editor/src/components/block-variation-picker/index.js]: Perhaps in __experimentalBlockVariationPicker

**Comment by richtabor** [comment]
A good step in the right direction, but we do need to also improve when the placeholders are smaller: 

CleanShot 2024-04-15 at 17 41 02

**Comment by jasmussen** [comment]
Good notes, thank you for the help! I've still on my radar to bring this home, hopefully I'll have some time tomorrow.

**Comment by t-hamano** [comment]
I'm in favor of refining the `Placeholder` component, but I am against removing the title from the `BlockVariationPicker` component and hiding the label.

This component has been around for a long time, but the title label didn't originally exist. Later, the missing label was raised as an issue in #18737, and the label was added in #19789.

As mentioned in #18737, the `BlockVariationPicker` component is also used by block developers, so there is no guarantee that the icon accurately represents the variation.

In this PR, can we focus on the following two things?

- Refine the Placeholder component itself
- For `BlockVariationPicker` components, only adjust the layout. In other words, title labels should remain visible in the Columns block.

Personally, I would also like to see labels added to the Group block variations. In other words, rather than displaying hard-coded variation buttons, we simply use the `BlockVariationPicker` components. This should also match the approach attempted in #60217.

**Comment by jasmussen** [comment]
Good feedback, all. I've been working to thread the needle, and based on your comments I took it to the drawing board again for a bit, and now have this:

mockup, wide and thin

This restores the title, addresses mobile, and I should be able to work this into the current PR. To make this work, designwise, I left-aligned again, but added padding that behaves almost as a max-width on the larger breakpoints. I'd still need to kind of tinker with the values in the PR, test it out, but let me know your thoughts.

One challenge is the equivalent of a typographical widow but for these items. It's probably possible to fix this using CSS grid, but it's somewhat more complex and I'd like to keep it separate. After all, this is how trunk looks still:
Screenshot 2024-04-19 at 11 22 16

**Comment by jameskoster** [comment]
Is the block name / icon supposed to be missing for Group? That's a bit unexpected.

I like the full-width buttons in smaller spaces.

**Comment by jasmussen** [comment]
> Is the block name / icon supposed to be missing for Group? That's a bit unexpected.

That's an existing change. This is trunk:

Screenshot 2024-04-23 at 08 09 32

We can bring it back, but I wonder if it should be in this PR, I'd honestly prefer it to be a separate one. The alternative is to remove icons from the other placeholders. Their value isn't entirely clear to me.

**Comment by jameskoster** [comment]
I see. I think I'd expect consistency either way. I suppose that's what @t-hamano is getting at... there are a few parts here:

* Refining the Placeholder.
* Refining BlockVariationPicker (to potentially include a prop for title display).
* Updating the icons for Columns / Groups.

It might be better to tackle them individually given the third party reliance on the components?

**Comment by t-hamano** [comment]
I would like to see the title label displayed in the Group block eventually, but I don't think it's necessary to deal with this in this PR.

Accessibility issues, including whether to add the title label to the Group block as well, may be addressed separately in #60217.

As mentioned in [this comment](https://github.com/WordPress/gutenberg/pull/59275#issuecomment-2064234766), how about focusing on the following points in this PR?

> - Refine the Placeholder component itself
> - For `BlockVariationPicker` components, only adjust the layout. In other words, title labels should remain visible in the Columns block.

**Comment by jasmussen** [comment]
I've added the titles locally but haven't pushed yet, have had some busy days. I hope to have time and energy to dive back into this tomorrow, and pending any other feedback, implement this design: https://github.com/WordPress/gutenberg/pull/59275#issuecomment-2066186771

Hopefull that gets us an i1, which I agree can and should be followed up with many more iterations.

**Comment by jasmussen** [comment]
Okay, I've an update for a re-review, please have a look. Here's a quick GIF of the branch:

status

* The design has moved close to [the sketch here](https://github.com/WordPress/gutenberg/pull/59275#issuecomment-2066186771), but not fully. Potentially we can move closer, but this was a good first reviewable and potentially landable step.
* That means left aligned across, that's simpler as far as maintaining back-compat.
* Improved status on narrow viewports with full-width buttons.
* 40px buttons applied across.
* Note that the missing title/icon on the Group variation is also bespoke changes existing in trunk that I've skipped touching for this branch.

Let me know your thoughts.

**Comment by jameskoster** [comment]
This feels pretty close, the 'regular' versions are looking good.

Some of the alignment details in the reduced versions looks a little strange to me. The main offender is the full-width buttons with left-aligned text – could that be centrally aligned? I wonder if it would be neater if the buttons in `BlockVariationPicker` 'fill' the space in a set number of columns so that padding is equidistant around them? Mockups:

Screenshot 2024-04-24 at 14 58 47

**Comment by jasmussen** [comment]
> could that be centrally aligned?

I actually tried that, just as you have in the sketch above, but I don't think it works, it breaks alignment with the surrounding pieces. I also ended up quite liking the left-aligned text, it matches these tab behaviors:

Screenshot 2024-04-24 at 17 01 18

Big scaling buttons for smaller sizes looks interesting, it could be something to explore separately. Though do note that those narrow context are _not_ mobile, they are mobile and if inserted into column blocks etc. — so it might be weird to see a mix of different sizes.

**Comment by jasmussen** [comment]
What do you think, should we land this?

**Comment by jasmussen** [comment]
Also, @t-hamano you voiced some specific concerns, when you have a moment can you see if they are addressed in this PR?

**Comment by jasmussen** [comment]
To remove a point of controversy, I centered the button text:

centered text

This is consistent with other fullwide buttons. I do think it can make sense to revisit that, but best to do holistically.

**Review by jameskoster — APPROVED** [review]
I'm not fully opposed to the left-aligned buttons, they just feel a bit unexpected compared with similar examples:

Screenshot 2024-04-25 at 13 33 08

Screenshot 2024-04-25 at 13 33 15

Screenshot 2024-04-25 at 13 33 26

Agree on exploring that separately. 

---

Otherwise this seems good to go in terms of design in core blocks. It may still be worth testing with block plugins making use of these components?

**Comment by t-hamano** [comment]
@jasmussen Thank you for the update! I will definitely check this PR tomorrow.

> It may still be worth testing with block plugins making use of these components?

I think this test is necessary. Many plugins use this component, some of which have a very large number of active installs. I'm going to test these as well.

[Search - WPdirectory](https://wpdirectory.net/search/01HWAMQQRC8TABG0VGD3D89SHM)

**Comment by richtabor** [comment]
A few comments: 

1. What about using 4px on all sides of the icon buttons here: 

| Current  | Proposed |
| ------------- | ------------- |
|CleanShot 2024-04-25 at 09 45 40|CleanShot 2024-04-25 at 09 44 14|


2. The featured image block's illustrated placeholder upload button is no longer centered: 
CleanShot 2024-04-25 at 09 43 10

**Comment by jasmussen** [comment]
Good catches, I'll address those!

**Comment by richtabor** [comment]
Overall, this is a good step forward. 

Thinking out loud here. As a post-merge question, I do wonder though what the ideal placeholder for layout placeholders specifically should be (and if we're there with this effort). 

For example, if I add a group block, do I need to see the group block label with instructions? Should the columns block placeholder look the same as the group block? You're choosing a layout; do you need to be informed to pick a layout—especially if we decided it's not necessary for group blocks? 

Would it matter if these looked the same? 

CleanShot 2024-04-25 at 09 50 46

File: packages/block-editor/src/components/block-variation-picker/content.scss [hunk 4]
```diff
@@ -1,89 +1,42 @@
...
-		padding: 0;
+	.components-button:hover {
+		background: none !important; // !important to simplify the selector.
 	}
 
-	&::before {
-		// Use `padding-bottom` trick to style element as perfect square.
-		content: "";
-		padding-bottom: 100%;
+	:hover svg {
```
Comment (t-hamano) [inline packages/block-editor/src/components/block-variation-picker/content.scss]: Would `.components-button:hover svg` be better?

File: packages/block-editor/src/components/block-variation-picker/content.scss [hunk 5]
```diff
@@ -1,89 +1,42 @@
...
+		align-items: center;
 	}
+}
 
-	&:last-child {
-		margin-right: 0;
-	}
+.block-editor-block-variation-picker__variations,
+.wp-block-group-placeholder__variations {
+	margin: 0 -4px; // Horizontally offset button padding to make icons line up.
```
Comment (t-hamano) [inline packages/block-editor/src/components/block-variation-picker/content.scss]: I'm wondering if we should apply this negative margin. I think this style depends on the width of the variation icons of the Group block and Column block being 48px. However, when a developer uses the `BlockVariationPicker` component, the path inside that icon does not necessarily have 48 width.

block-variation-picker-negative-margin

How about applying this negative margin only to the Group block and the Column block?

File: packages/components/src/placeholder/style.scss [hunk 6]
```diff
@@ -38,15 +39,14 @@
 }
 
 .components-placeholder__label {
-	display: flex;
 	font-weight: 600;
-	margin-bottom: $grid-unit-20;
 	align-items: center;
+	display: inline-flex;
```
Comment (t-hamano) [inline packages/components/src/placeholder/style.scss]: What is the reason for changing from `flex` to `inline-flex`?

File: packages/block-editor/src/components/media-placeholder/content.scss [hunk 7]
```diff
@@ -29,10 +29,6 @@
 	flex-shrink: 1;
 }
 
-.block-editor-media-placeholder__button {
-	margin-bottom: 0.5rem;
-}
```
Comment (t-hamano) [inline packages/block-editor/src/components/media-placeholder/content.scss:34]: Due to this change, we should also be able to remove the margin reset style here:

https://github.com/WordPress/gutenberg/blob/e8c35082189781851544e09a326d32a1b29579e5/packages/block-editor/src/components/media-placeholder/content.scss#L4

image-url-margin

**Review by t-hamano — COMMENTED** [review]
I looked into the parts this PR is related to. I think a few things need to be addressed in order for this PR to ship. Sorry for including so many screenshots 😅

### BlockVariationPicker
I'm a little concerned about the button being adjacent to the label when focused. How about adding a margin of around 4px above the label?

block-variation-picker-spacing

## Blocks

### Embeds Block

The layout is a little broken.

| trunk | This PR |
|--------|--------|
| embeds-block-before | embeds-block-after | 

Also, this margin is probably unnecessary.

https://github.com/WordPress/gutenberg/blob/72afd4e889d754b5a7d7fc66205fd259ca077df3/packages/block-library/src/embed/editor.scss#L21-L22

### Image Block, Video Block

The buttons are centered. This is due to the style [here](https://github.com/WordPress/gutenberg/pull/59275/files#diff-e6d39e2bf2050a2e7a7022d39b3ddb498bf87165293fd8759302bab4c2dfb346R159). This style is only needed for the Site Logo block and Featured Image block, so I think it's better to define it as an editor style for the individual blocks rather than for the `Placehoder` component itself.

video-block




### Post Featured Image Block

In mobile view, the upload button is shifted to the left.

featured-image-block-mobile

### Site Logo Block

The upload button is shifted to the left.

site-logo-block

## Block Plugins

I've tested some plugins that have a very high number of installs, and they don't seem to have a big impact.

<details><summary>Screenshots</summary>

### JetPack - Form Block

plugin-jetpack-form

plugin-jetpack-form-mobile

### CoBlocks - Carousel

coblocks-carousel

### Otter Blocks - Popup

otter-blocks-popup

</details>

## Others

I think these can be done as a follow-up, but we may also want to consider them as the next step.

- Elements with the `components-placeholder__fieldset` class are always rendered, resulting in unnecessary white space if the `Placeholder` component has no children. If there are no children, the element itself should not be rendered.

placeholder-spacing


- Why hide instructions in the mobile viewport?

- Several blocks contain CSS that overrides the default styles of placeholders. Some of these should be able to be removed after this PR is applied.

File: packages/block-editor/src/components/block-variation-picker/content.scss (hunk 4 above)
Comment (jasmussen) [reply packages/block-editor/src/components/block-variation-picker/content.scss]: Yes! Done.

File: packages/block-editor/src/components/block-variation-picker/content.scss (hunk 5 above)
Comment (jasmussen) [reply packages/block-editor/src/components/block-variation-picker/content.scss]: Removed.

File: packages/components/src/placeholder/style.scss (hunk 6 above)
Comment (jasmussen) [reply packages/components/src/placeholder/style.scss]: That was an older iteration. Good catch, changed it back.

File: packages/block-editor/src/components/media-placeholder/content.scss (hunk 7 above)
Comment (jasmussen) [reply packages/block-editor/src/components/media-placeholder/content.scss:34]: Done.

**Comment by jasmussen** [comment]
Thank you for the in-depth feedback 🙏 — I have pushed fixes and hope I do it justice. At a high level, here it should all be captured:

status

I would really like to land this because so many opportunities for improvement appear when touching this, but I also would love to keep the PR as focused as possible so as to not balloon. Just one example, here's the Embed block URL:
Screenshot 2024-04-26 at 12 39 52

That right margin makes the gap between input and button look wrong. It comes from here:
Screenshot 2024-04-26 at 12 40 28

That feels a bit too opinionated. But I also don't want to touch that as part of this PR, ideally, to keep things as on-topic as possible.

For that same reason, I kept the Embed improvements a separate commit. It now looks like this:

Screenshot 2024-04-26 at 12 45 01

I used VStack and HStack to keep things simple. Hope that's reasonable.

**Comment by jasmussen** [comment]
I wanted to touch on this separately because there's an important nuance.

> Why hide instructions in the mobile viewport?

It is not hidden on the mobile viewport. It is hidden when the placeholder becomes small. So there's a JS based element query at play here that applies `is-medium` and `is-small` CSS classes based on the width of the placeholder. 

This is because Placeholder states can be as small as this:

site logo

We are so used to seeing the big wide placeholder patterns that it's tempted to add icons, titles, descriptions, notices, error states, and more. But the fact is that it's increasingly common to have patterns that put these blocks with placeholder states in extremely narrow contexts. Here's one from Twenty Twenty-Four:

pattern

That's from this branch, and it's revealing some brokenness that I'm now diving in to fix. 

But I'm sharing because the fact that blocks with placeholder states can become as broken as they are in trunk, when used in narrow contexts, is the entire and full motivation for why this PR initially removed far more than what we ended up with here. The unfortunate fact is that the Placeholder pattern did not scale as it was initially intended. 

<details><summary>Test pattern</summary>

```
<!-- wp:columns {"style":{"spacing":{"blockGap":{"top":"0","left":"var:preset|spacing|40"},"margin":{"top":"0","bottom":"0"}}}} -->
<div class="wp-block-columns" style="margin-top:0;margin-bottom:0"><!-- wp:column {"style":{"spacing":{"blockGap":"0"}}} -->
<div class="wp-block-column"><!-- wp:group {"style":{"spacing":{"blockGap":"var:preset|spacing|50"}},"layout":{"type":"constrained"}} -->
<div class="wp-block-group"><!-- wp:image -->
<figure class="wp-block-image"><img alt=""/></figure>
<!-- /wp:image -->

<!-- wp:image -->
<figure class="wp-block-image"><img alt=""/></figure>
<!-- /wp:image -->

<!-- wp:image -->
<figure class="wp-block-image"><img alt=""/></figure>
<!-- /wp:image --></div>
<!-- /wp:group --></div>
<!-- /wp:column -->

<!-- wp:column {"style":{"spacing":{"blockGap":"0","padding":{"top":"0"}}}} -->
<div class="wp-block-column" style="padding-top:0"><!-- wp:spacer {"height":"var:preset|spacing|50"} -->
<div style="height:var(--wp--preset--spacing--50)" aria-hidden="true" class="wp-block-spacer"></div>
<!-- /wp:spacer -->

<!-- wp:group {"style":{"spacing":{"blockGap":"var:preset|spacing|50"}},"layout":{"type":"constrained"}} -->
<div class="wp-block-group"><!-- wp:image -->
<figure class="wp-block-image"><img alt=""/></figure>
<!-- /wp:image -->

<!-- wp:image -->
<figure class="wp-block-image"><img alt=""/></figure>
<!-- /wp:image -->

<!-- wp:image -->
<figure class="wp-block-image"><img alt=""/></figure>
<!-- /wp:image --></div>
<!-- /wp:group -->

<!-- wp:spacer {"height":"var:preset|spacing|50"} -->
<div style="height:var(--wp--preset--spacing--50)" aria-hidden="true" class="wp-block-spacer"></div>
<!-- /wp:spacer --></div>
<!-- /wp:column -->

<!-- wp:column {"style":{"spacing":{"blockGap":"0"}}} -->
<div class="wp-block-column"><!-- wp:group {"style":{"spacing":{"blockGap":"var:preset|spacing|50"}},"layout":{"type":"constrained"}} -->
<div class="wp-block-group"><!-- wp:image -->
<figure class="wp-block-image"><img alt=""/></figure>
<!-- /wp:image -->

<!-- wp:image -->
<figure class="wp-block-image"><img alt=""/></figure>
<!-- /wp:image -->

<!-- wp:image -->
<figure class="wp-block-image"><img alt=""/></figure>
<!-- /wp:image --></div>
<!-- /wp:group --></div>
<!-- /wp:column -->

<!-- wp:column {"style":{"spacing":{"blockGap":"0","padding":{"top":"0"}}}} -->
<div class="wp-block-column" style="padding-top:0"><!-- wp:spacer {"height":"var:preset|spacing|50"} -->
<div style="height:var(--wp--preset--spacing--50)" aria-hidden="true" class="wp-block-spacer"></div>
<!-- /wp:spacer -->

<!-- wp:group {"style":{"spacing":{"blockGap":"var:preset|spacing|50"}},"layout":{"type":"constrained"}} -->
<div class="wp-block-group"><!-- wp:image -->
<figure class="wp-block-image"><img alt=""/></figure>
<!-- /wp:image -->

<!-- wp:image -->
<figure class="wp-block-image"><img alt=""/></figure>
<!-- /wp:image -->

<!-- wp:image -->
<figure class="wp-block-image"><img alt=""/></figure>
<!-- /wp:image --></div>
<!-- /wp:group -->

<!-- wp:spacer {"height":"var:preset|spacing|50"} -->
<div style="height:var(--wp--preset--spacing--50)" aria-hidden="true" class="wp-block-spacer"></div>
<!-- /wp:spacer --></div>
<!-- /wp:column --></div>
<!-- /wp:columns -->
```


</details>

**Comment by jasmussen** [comment]
I fixed an issue with site logo centering, and further polished very narrow contexts. Here's thin columns:

Screenshot 2024-04-26 at 13 18 00

Yes, that's not ideal, but here's trunk:

Screenshot 2024-04-26 at 13 20 28

I do not believe there's a way to address those contexts without either moving items into a modal opened by a button, or more generally moving towards [this](https://github.com/WordPress/gutenberg/pull/59275#issuecomment-2077267680).

The PR should now be ready for any final thoughts. Thanks again for all your outstanding feedback. Thank you!

**Review by t-hamano — APPROVED** [review]
LGTM! I believe it now looks correct in all scenarios.

Regarding blocks with the placeholder form, such as the Embed block, #41961 should continue to improve them.

I don't think there is currently a perfect solution to the problem, which is a very narrow context. But at least this PR is better than the trunk.

**Comment by jasmussen** [comment]
Any concerns about merging on a Friday? Happy to hold off until Monday if you all prefer!

**Comment by t-hamano** [comment]
I think it's okay to merge now 🚀

**Comment by jasmussen** [comment]
Thanks all. I'll look at followups, and will keep monitoring feedback on this PR as well.

**Comment by afercia** [comment]
While I do realize the huge effort behind this PR, I have to note that it introduced a regression as the Columns placeholder now uses the 'light gray' color for the icons that was already reported as not accessible for the Group block in https://github.com/WordPress/gutenberg/issues/60206

Also, removing visible labels and text is not OK for accessibility, and I would say for general usability as well.

**Comment by joedolson** [comment]
I need to second @afercia's comments. This update has replicated a clear and unambiguous existing accessibility issue, since now both the Group and Columns placeholders fail WCAG 1.4.11. 

See: [WCAG 1.4.11: Non-text contrast](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast)

**Pull request merged**